iati/core/tests/test_rulesets.py

```diff
@@ -222,87 +237,113 @@ def rule_constructor(self, rule_type):
-    #     """Check that a given Rule returns the expected result when given a Dataset."""
-    #     assert not rule_is_valid_for_case.is_valid_for(invalid_dataset)
+    def test_is_invalid_for(self, invalid_dataset, rule_invalid):
+        """Check that a given Rule returns the expected result when given a Dataset."""
+        assert not rule_invalid.is_valid_for(invalid_dataset)
 
     @pytest.mark.parametrize("junk_data", iati.core.tests.utilities.find_parameter_by_type([], False))
-    def test_is_valid_for_raises_error_on_non_permitted_argument(self, rule_basic_init, junk_data):
+    def test_is_valid_for_raises_error_on_non_permitted_argument(self, rule_instantiating, junk_data):
         """Check that a given Rule returns expected error when passed an argument that is not a Dataset."""
         with pytest.raises(AttributeError):
-            rule_basic_init.is_valid_for(junk_data)
+            rule_instantiating.is_valid_for(junk_data)
 
-    def test_is_valid_for_raises_error_when_passed_an_etree(self, rule_basic_init):
+    def test_is_valid_for_raises_error_when_passed_an_etree(self, rule_instantiating):
         """Check that an error is raised if an etree is given as an argument instead of a Dataset.
 
         Todo:
             Use more generic Dataset.
 
         """
         with pytest.raises(AttributeError):
-            rule_basic_init.is_valid_for(iati.core.resources.load_as_tree(iati.core.resources.get_test_data_path('valid_atleastone')))
+            rule_instantiating.is_valid_for(iati.core.resources.load_as_tree(iati.core.resources.get_test_data_path('valid_atleastone')))
+
+    def test_multiple_valid_context_matches_is_valid_for(self, valid_multiple_context, valid_nest_case, rule_constructor, valid_dataset):
+        """Check Rule returns expected result when checking multiple contexts."""
+        rule = rule_constructor(valid_multiple_context, valid_nest_case)
+        assert rule.is_valid_for(valid_dataset)
+
+    def test_multiple_valid_context_matches_is_invalid_for(self, valid_multiple_context, invalid_nest_case, rule_constructor, invalid_dataset):
+        """Check Rule returns expected result when checking multiple contexts."""
+        rule = rule_constructor(valid_multiple_context, invalid_nest_case)
+        assert not rule.is_valid_for(invalid_dataset)
 
-    def test_empty_xpath_base_rule_init_normalised_paths(self, rule_empty_xpath_base):
-        """Check that a Rule with an empty xpath_base can be instantiated correctly with normalised `paths`.
+    def test_condition_case_is_True(self, valid_condition_rule, valid_dataset):
+        """Check that if a condition is `True`, the rule returns None which is considered equivalent to skipping."""
+        assert valid_condition_rule.is_valid_for(valid_dataset) is None
+
+    def test_condition_case_is_False(self, invalid_condition_rule, invalid_dataset):
+        """Check that if a condition is `False`, the rule validates normally.
```

Comment:
Wrt https://github.com/IATI/iati.core/pull/104#discussion_r136311705

It's necessary to check both the valid and invalid dataset here because otherwise all it's checking is that if the condition is `False`, it returns `False`. This isn't quite the same thing.